Main feature

chiztechnology opened the pull request:
### React capstone project - Metrics webapp

In this pull request, we worked on the main features of a Weather App working with WEATHER API.

- the first page contains a list of different town/cities and 
- the second-page displays details according to the selected item.

<img width="1440" alt="Screenshot 2023-02-18 at 22 14 51" src="https://user-images.githubusercontent.com/68948316/219904864-697e60a2-b99c-41a3-8a42-b055997dc8fa.png">


### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config). ✅ 
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md). ✅ 
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md). ✅ 

### HTML/CSS & JavaScript requirements

- Follow our list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

chiztechnology committed:
Init react app

chiztechnology committed:
Install dependencies

chiztechnology committed:
Add babel and eslint config files

chiztechnology committed:
Remove broken readme file

chiztechnology committed:
Set correct readme file

chiztechnology committed:
Add logo to readme file header

chiztechnology committed:
Set Mantine libraby for building UI

chiztechnology committed:
Set up react router in the app

chiztechnology committed:
Create left navbar design

chiztechnology committed:
Customize nav header

chiztechnology committed:
Set styling to components

chiztechnology committed:
Integrate redux to the app

chiztechnology committed:
Fix design of cities item

chiztechnology committed:
Set sample array for testing design

chiztechnology committed:
Work on city weather detail

chiztechnology committed:
Set design of city weather details

chiztechnology committed:
Fix redux issue annd clear useless code

chiztechnology committed:
Fix design responsiveness

chiztechnology committed:
Add loom video link to readme file

chiztechnology committed:
Add live demo link to readme file

chiztechnology committed:
Set app ready for production

chiztechnology committed:
Set React testinng environment

chiztechnology committed:
Add linters to project

chiztechnology committed:
Fix package json config

chiztechnology committed:
Clear css code

chiztechnology committed:
Fix styling issues

chiztechnology committed:
Clear js code

chiztechnology committed:
Clear css code

anagudelogu commented on apikey.txt:
```diff
@@ -0,0 +1 @@
+key='45ef86c3a51d4ca7acc85658231502'
```
- [ ] **[OPTIONAL]:** Since API keys are sensitive data, I suggest you use env variables to store them! you should never push sensitive data into public remote repositories. You can read more about [env variables here](https://www.architect.io/blog/2022-08-16/react-environment-variables-developers-guide/). 🤓 🔧

anagudelogu commented on src/App.js:
```diff
@@ -0,0 +1,21 @@
+  // const dispatch = useDispatch();
+  // useEffect(() => {
+  //   dispatch(getWeatherFromRandomCities());
+  // }, [dispatch]);
```
- [ ] I noticed that you have a lot of commented lines of code across your code base. Please make sure that you remove those as they make your code look dirty and reduce its readability. 🤓 🔧

anagudelogu commented on src/components/Cities.js:
```diff
@@ -0,0 +1,58 @@
+    id: 3, name: 'kinshasa', country: 'DR Congo', localtime: '1:09 PM', humidity: '27%', temp_c: '32',
+  }, {
+    id: 4, name: 'san francisco', country: 'United States', localtime: '8:09 PM', humidity: '18%', temp_c: '15',
+  }];
```
- [ ] The goal of this project is to consume an external API and get this data. Please avoid using static data in your components. 🤓 🔧

anagudelogu commented on src/components/Cities.js:
```diff
@@ -0,0 +1,58 @@
+        searchable
+        data={autocompleteArray}
+        style={{ width: 420 }}
+      />
```
- [ ] Also, the purpose of the "Search" component is to filter the data present in the UI, at the moment you have a select box but its selection is not working. Please consider refactoring your component to be a search box and every time you type a letter it filters the results in your UI. 🤓 🔧 

![image](https://user-images.githubusercontent.com/77069011/219906035-2ec5097a-d5e4-4668-a997-59aaa426a28d.png)

anagudelogu commented on src/components/CityItem.js:
```diff
@@ -0,0 +1,40 @@
+
+  // const { humididy, temp_c } = props.data ? props.data.condition: 0;
+  // const { name, country, localtime } = '';
+  // const { humididy, temp_c } = '';
```
- [ ] Again, you're having a lot of commented code, please remove it! 🤓 🔧

anagudelogu commented on src/components/CityItem.js:
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable */
```
- [ ] Please avoid disabling linters in your code, they serve as a way to ensure a consistent format and the use of best practices across your codebase, instead, consider fixing the errors you're getting. 🤓 🔧

anagudelogu commented on src/components/Weather.js:
```diff
@@ -0,0 +1,60 @@
+        {/* Your application here */}
+        <Routes>
+          <Route path="/" element={<Cities />} />
+          <Route path="/details" element={<Details />} />
```
- [ ] The details page is meant to be unique for each data element you're getting from the external API. To achieve this, you can use URL params with some unique detail, e.g. `id` or `name`. Something like this:

```suggestion
          <Route path="/details/:id" element={<Details />} />
```

And then, you modify the link you're using when the user clicks a city to include this parameter 🤓 🔧

anagudelogu commented on src/components/Details.js:
```diff
@@ -0,0 +1,128 @@
+            fontSize: 14,
+            '&:hover': {
+              backgroundColor: theme.fn.darken('#4b4b49', 0.05),
+            },
```
- [ ] Please avoid using inline styles, they make your code less readable and make your components look dirty! According to the [HTML and CSS best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md):

![image](https://user-images.githubusercontent.com/77069011/219906318-8c33e491-a83d-469b-ae45-28f4e68ec7c5.png)

anagudelogu commented on src/redux/weather/DetailWeather.js:
```diff
@@ -0,0 +1,49 @@
+
+export const getWeatherDetails = createAsyncThunk(FETCH, async (name) => {
+  const WeatherDetails = await axios.get(`http://api.weatherapi.com/v1/current.json?key=${apiKey}&q=${name}`);
+  return WeatherDetails.data;
```
- [ ] Please consider using the right naming conventions for variables, here you're using PascalCase which is reserved for classes/components. Instead, consider using camelCase for variables! 🤓 🔧 


```suggestion
  const weatherDetails = await axios.get(`http://api.weatherapi.com/v1/current.json?key=${apiKey}&q=${name}`);
  return weatherDetails.data;
```

anagudelogu commented on src/redux/weather/weatherSlice.js:
```diff
@@ -0,0 +1,35 @@
+  weather: [],
+};
+
+export const getWeatherFromRandomCities = createAsyncThunk(FETCH, async () => {
+  const weather = [{
+    id: 1, name: 'london', country: 'United Kingdom', localtime: '12:09 PM', humidity: '80%', temp_c: '23',
+  }, {
+    id: 2, name: 'paris', country: 'France', localtime: '1:09 PM', humidity: '10%', temp_c: '12',
+  }, {
+    id: 3, name: 'kinshasa', country: 'DR Congo', localtime: '1:09 PM', humidity: '27%', temp_c: '32',
+  }, {
+    id: 4, name: 'san francisco', country: 'United States', localtime: '8:09 PM', humidity: '18%', temp_c: '15',
+  }];
+  return weather;
+});
```
- [ ] You're not interacting with an external API as required by the project, please don't seed static data to your app like this, you're not meeting one of the most important goals of this activity which is to consume APIs in your React/Redux apps. 🤓 🔧

anagudelogu reviewed (changes requested):
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @chiztechnology, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 

<img src="https://media2.giphy.com/media/3ohhwi25ISXC7Z4tMs/giphy-downsized-medium.gif"/>

## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

- [ ] You did a great job working on this milestone, however, you didn't use the required design guidelines. According to the requirements:

![image](https://user-images.githubusercontent.com/77069011/219906500-a0cd432e-a9fd-4baa-9c51-224007c31d8d.png)

But you have an entirely different design than the one provided by [Nelson Sakwa](https://www.behance.net/sakwadesignstudio). Please consider using the correct design. 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

chiztechnology commented:
hey, @anagudelogu how am I going to follow the design guidelines while working on another kind of project as you see I've to make the components customized according to my topic.
and the reference design is a mobile app. I'm building a web app.
how to handle that?

I've been struggling about that during 3 days and I didn't find any solution

anagudelogu commented:
Hi @chiztechnology! thanks for reaching out. I think that you can structure the content of your app in the given design, if you see, you have a list of cities that have weather data, right? 

So let's say this is the cities view (home page):

![image](https://user-images.githubusercontent.com/77069011/221191134-fd105c73-6dcf-4ba7-b70d-0e30f4069b6e.png)

And, when the user clicks on the city, you want to redirect to a details page where you show the weather information for that specific city, so this view could be designed similarly (no graphs needed) to this view:

![image](https://user-images.githubusercontent.com/77069011/221191613-2fe6b7ce-088e-4590-b0b9-44e1aca20cee.png)


So following the project requirements:

![image](https://user-images.githubusercontent.com/77069011/221191736-80795d99-3dd1-4d9c-b014-c82293ef593e.png)

You can choose between blue or pink colors, use a similar if not the same font, and layout your page similar to the Behance design! 🤓 🚀

chiztechnology commented:
Hey ! So I finally make a mobile phone design ? Means when somebody opens
my website it will display a mobile size app on the wide screen ? Or it has
to fit on the screen width ?

On Fri, Feb 24, 2023 at 3:36 PM Andres Agudelo Guzman <
***@***.***> wrote:

> Hi @chiztechnology <https://github.com/chiztechnology>! thanks for
> reaching out. I think that you can structure the content of your app in the
> given design, if you see, you have a list of cities that have weather data,
> right?
>
> So let's say this is the cities view (home page):
>
> [image: image]
> <https://user-images.githubusercontent.com/77069011/221191134-fd105c73-6dcf-4ba7-b70d-0e30f4069b6e.png>
>
> And, when the user clicks on the city, you want to redirect to a details
> page where you show the weather information for that specific city, so this
> view could be designed similarly (no graphs needed) to this view:
>
> [image: image]
> <https://user-images.githubusercontent.com/77069011/221191613-2fe6b7ce-088e-4590-b0b9-44e1aca20cee.png>
>
> So following the project requirements:
>
> [image: image]
> <https://user-images.githubusercontent.com/77069011/221191736-80795d99-3dd1-4d9c-b014-c82293ef593e.png>
>
> You can choose between blue or pink colors, use a similar if not the same
> font, and layout your page similar to the Behance design! 🤓 🚀
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chiztechnology/weather-app/pull/1#issuecomment-1443691924>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQOBCXETZWVMFKGP5Q52OYDWZC2OXANCNFSM6AAAAAAVAT5YX4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

anagudelogu commented:
@chiztechnology The main goal is to make a website using the mobile-first approach (remember the portfolio? mobile-first) but only mobile styles are required for this Milestone. After the requirement is met, is optional (but highly recommended) to create responsive styles for other screen sizes! 🚀 🤓

chiztechnology commented:
Owhhh I understand. Thanks a lot dear.
I remember the mobile first approch.

Im really happy 😊 thanks a lot.
I correct the bugs and refactor the design and I’ll submit for another
review

On Fri, Feb 24, 2023 at 6:19 PM Andres Agudelo Guzman <
***@***.***> wrote:

> The main goal is to make a website using the mobile-first approach
> (remember the portfolio? mobile-first) but only mobile styles are required
> for this Milestone. After the requirement is met, is optional (but highly
> recommended) to create responsive styles for other screen sizes! 🚀 🤓
>
> Hey ! So I finally make a mobile phone design ? Means when somebody opens
> my website it will display a mobile size app on the wide screen ? Or it has
> to fit on the screen width ?
> … <#m_-1599631171207205789_>
> On Fri, Feb 24, 2023 at 3:36 PM Andres Agudelo Guzman < *@*.*> wrote: Hi
> @chiztechnology <https://github.com/chiztechnology>
> https://github.com/chiztechnology <https://github.com/chiztechnology>!
> thanks for reaching out. I think that you can structure the content of your
> app in the given design, if you see, you have a list of cities that have
> weather data, right? So let's say this is the cities view (home page):
> [image: image]
> https://user-images.githubusercontent.com/77069011/221191134-fd105c73-6dcf-4ba7-b70d-0e30f4069b6e.png
> <https://user-images.githubusercontent.com/77069011/221191134-fd105c73-6dcf-4ba7-b70d-0e30f4069b6e.png>
> And, when the user clicks on the city, you want to redirect to a details
> page where you show the weather information for that specific city, so this
> view could be designed similarly (no graphs needed) to this view: [image:
> image]
> https://user-images.githubusercontent.com/77069011/221191613-2fe6b7ce-088e-4590-b0b9-44e1aca20cee.png
> <https://user-images.githubusercontent.com/77069011/221191613-2fe6b7ce-088e-4590-b0b9-44e1aca20cee.png>
> So following the project requirements: [image: image]
> https://user-images.githubusercontent.com/77069011/221191736-80795d99-3dd1-4d9c-b014-c82293ef593e.png
> <https://user-images.githubusercontent.com/77069011/221191736-80795d99-3dd1-4d9c-b014-c82293ef593e.png>
> You can choose between blue or pink colors, use a similar if not the same
> font, and layout your page similar to the Behance design! — Reply to this
> email directly, view it on GitHub <#1 (comment)
> <https://github.com/chiztechnology/weather-app/pull/1#issuecomment-1443691924>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AQOBCXETZWVMFKGP5Q52OYDWZC2OXANCNFSM6AAAAAAVAT5YX4
> <https://github.com/notifications/unsubscribe-auth/AQOBCXETZWVMFKGP5Q52OYDWZC2OXANCNFSM6AAAAAAVAT5YX4>
> . You are receiving this because you were mentioned.Message ID: @.*>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chiztechnology/weather-app/pull/1#issuecomment-1443941300>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQOBCXH4PX5IU4UBHCTSL33WZDNQ7ANCNFSM6AAAAAAVAT5YX4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

chiztechnology committed:
Set up dotenv variable for apiKey,clear code and use api calls

chiztechnology committed:
Clear city item component

chiztechnology committed:
Refactor design in mobile-first approach

chiztechnology committed:
Remove unused commented  code

chiztechnology committed:
Clear css code

chiztechnology committed:
Fix space indentation

chiztechnology committed:
clear code

chiztechnology commented:
Hello dear.
thank you for your message.
I refactored the design and corrected everything as requested 😊 🎊  ✅ 
<img width="1440" alt="Screenshot 2023-02-25 at 11 08 45" src="https://user-images.githubusercontent.com/68948316/221349169-b0efd9b8-57e9-40b8-b848-e7d889c8a1ef.png">
<img width="1440" alt="Screenshot 2023-02-25 at 11 08 50" src="https://user-images.githubusercontent.com/68948316/221349190-5ecc1ea1-5261-41d7-93c1-0ae563eb3a38.png">

chiztechnology commented:
I submit for another review now

christianbiring1 commented on src/components/Details.js:
```diff
@@ -0,0 +1,106 @@
+        >
+          <FaArrowLeft color="white" />
+        </UnstyledButton>
+        <Text className="city-header-title">Town/Cities</Text>
```
- [ ] Kindly, as suggested by the previous reviewer, make sure for each city, once you have its corresponding details. as you may notice with me, once the user tries to see the details for cities they get the exact same results. and the error is coming from your `API endpoint.` as you can see in the console :point_down: 
![image](https://user-images.githubusercontent.com/97100091/221350789-dfe9ff8b-ab04-49ed-baea-856904d8991d.png)
Kindly make sure the details of each city are retrieved correctly and you get their data.

christianbiring1 commented on src/components/CityItem.js:
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable */
```
- [ ] Also as mentioned previously, kindly consider looking for a way to correct your linters errors instead of disabling them for they help standardize your codebase and make it cleaner.

christianbiring1 commented on src/components/Weather.js:
```diff
@@ -0,0 +1,25 @@
+  }, []);
+
+  return (
+    <BrowserRouter>
+      {/* Your application here */}
+      <Routes>
+        <Route path="/" element={<Cities />} />
+        <Route path="/details" element={<Details />} />
+      </Routes>
+    </BrowserRouter>
```
- [ ] As mentioned previously, kindly make sure you are using proper `Routing` on your `details` to make sure the user gets the details for each specific `city` based on it `id` or `name.` i.e you may say;
```jsx
<Route path="/details/:id(name)" element={<Details />} />
// And make sure the use navigate is used correctly.
```
You may consider check [here](https://www.geeksforgeeks.org/reactjs-usenavigate-hook/) to see how the `UseNvagivate` hook works

christianbiring1 reviewed (changes requested):
Hi @chiztechnology , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
# STATUS: REQUEST CHANGES :recycle: 
![soclose](https://user-images.githubusercontent.com/97100091/221351732-8ef04985-2402-43e2-8809-b622d3354cb0.gif)

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

chiztechnology committed:
Fix navigation with params

chiztechnology commented:
hello dear @christianbiring1 thank you for your great review.
I Just fixed everything as requested.

Best regards 😊 🎊  ✅ 

<img width="1440" alt="Screenshot 2023-02-25 at 17 44 40" src="https://user-images.githubusercontent.com/68948316/221366190-7a0e3d59-65bc-43ba-a864-8e1c4203d89c.png">
<img width="1440" alt="Screenshot 2023-02-25 at 17 44 51" src="https://user-images.githubusercontent.com/68948316/221366195-1d7753f1-2bda-423c-a1cd-b758e4904e60.png">

t-yanick commented on src/components/Cities.js:
```diff
@@ -0,0 +1,56 @@
+        />
+        <Text className="weather-desc-title">Weather App</Text>
+      </div>
+
```
- [ ] Great job working on some of the changes suggested by the previous reviewer. Your design is now looking like the given [Nelson Sakwa design](https://www.behance.net/gallery/31579789/Ballhead-App-%28Free-PSDs%29). Unforf tunately the previous reviewer spoke of select box with selection not working, but right now there is even no search box for any `filter`. 
![filter](https://user-images.githubusercontent.com/77069011/219906035-2ec5097a-d5e4-4668-a997-59aaa426a28d.png) 
See screenshot below, there is no search bar to filter maybe by city name. Kindly review this. 
![Screen Shot 2023-02-25 at 8 13 12 AM](https://user-images.githubusercontent.com/62849872/221367493-da717be4-592b-45b1-b080-de159f1042fb.png)

- [ ] When I click on a city to get details about the city, your design presents some weird dots and brackets which is not found on the given design. Kindly get this cleaned. See screenshot below;
![Screen Shot 2023-02-25 at 8 16 03 AM](https://user-images.githubusercontent.com/62849872/221367770-f2bc12f4-8629-4e7a-9702-5804f709c430.png)

t-yanick reviewed (changes requested):
Hi @chiztechnology , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

chiztechnology commented:
Thank you ! Let me update the app and submit for another review

On Sat, Feb 25, 2023 at 6:26 PM Tazoh Yanick ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @chiztechnology <https://github.com/chiztechnology> ,
>
> Great work on making the changes requested by a previous reviewer 👏
> You've done well implementing some of the requested changes, but there are
> still some which aren't addressed yet.
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix won't stop the approval of this
> PR. However, I strongly recommend you to take them into account as they can
> make your code better. Some of them were simply missed by the previous
> reviewer and addressing them will really improve your application.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In src/components/Cities.js
> <https://github.com/chiztechnology/weather-app/pull/1#discussion_r1117944499>
> :
>
> > +      {/* header status bar */}
>
> +      <div className="cities-status-bar">
>
> +        <Text className="city-header-title">My Weather App</Text>
>
> +      </div>
>
> +
>
> +      {/* sub-header */}
>
> +      <div className="sub-header-city">
>
> +        <Image
>
> +          radius="md"
>
> +          src="https://cdn2.iconfinder.com/data/icons/weather-flat-14/64/weather02-512.png"
>
> +          alt="Weather icon"
>
> +          width={80}
>
> +        />
>
> +        <Text className="weather-desc-title">Weather App</Text>
>
> +      </div>
>
> +
>
>
>
>    -
>
>    Great job working on some of the changes suggested by the previous
>    reviewer. Your design is now looking like the given Nelson Sakwa design
>    <https://www.behance.net/gallery/31579789/Ballhead-App-%28Free-PSDs%29>.
>    Unforf tunately the previous reviewer spoke of select box with selection
>    not working, but right now there is even no search box for any filter.
>    [image: filter]
>    <https://user-images.githubusercontent.com/77069011/219906035-2ec5097a-d5e4-4668-a997-59aaa426a28d.png>
>    See screenshot below, there is no search bar to filter maybe by city
>    name. Kindly review this.
>    [image: Screen Shot 2023-02-25 at 8 13 12 AM]
>    <https://user-images.githubusercontent.com/62849872/221367493-da717be4-592b-45b1-b080-de159f1042fb.png>
>    -
>
>    When I click on a city to get details about the city, your design
>    presents some weird dots and brackets which is not found on the given
>    design. Kindly get this cleaned. See screenshot below;
>    [image: Screen Shot 2023-02-25 at 8 16 03 AM]
>    <https://user-images.githubusercontent.com/62849872/221367770-f2bc12f4-8629-4e7a-9702-5804f709c430.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chiztechnology/weather-app/pull/1#pullrequestreview-1314538510>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQOBCXDXVHUMG6OJ5VE2VFLWZIXB3ANCNFSM6AAAAAAVAT5YX4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

chiztechnology committed:
Create search box

chiztechnology commented:
Hello dear @t-yanick  I just fixed the broken detail screen design and added a search box.
Kindly check and let me know if anything requires improvements.

Thanks 🤝 

<img width="1440" alt="Screenshot 2023-02-25 at 20 08 30" src="https://user-images.githubusercontent.com/68948316/221372885-4af93126-821a-4d0e-9490-60046cde5f77.png">

t-yanick commented:
@chiztechnology great job. Kindly request for a re-review.

TadesseDev commented on src/components/Cities.js:
```diff
@@ -1,20 +1,32 @@
 } from '@mantine/core';
 import { useSelector } from 'react-redux';
+import { FaSearch } from 'react-icons/fa';
 import CityItem from './CityItem';
 import './cities.css';
 
 const Cities = () => {
+  const { cities } = useSelector((state) => state.cities);
   const [loading, setLoading] = useState(true);
+  const [filteredData, setFilteredData] = useState([]);
+
   useEffect(() => {
-    setTimeout(() => {
+    setLoading(true);
+    // setTimeout(() => {
+    if (cities.length > 0) {
       setLoading(false);
-    }, 500);
-  }, []);
+    }
+    setFilteredData(cities);
+    // }, 1000);
+  }, [cities]);
 
-  const { cities } = useSelector((state) => state.cities);
+  const handleChange = (text) => {
+    const content = text.target.value;
+    console.log(content);
+    setFilteredData(cities.filter((e) => e.city.toLowerCase().includes(content.toLowerCase())));
+  };
 
```
- [ ] Dear @chiztechnology, Good job making the requested changes. But while doing this, I notice you have commented on old codes. And it's among the primary requirement to keep your codebase clean. Please consider removing commented codes as it will make your codebase messy, Remember if you ever like to see the codes you have written once you can use your commit history. Check this list of [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) for more on this.
- [Optional] Also, I notice your app works fine locally but fails on the live server. The last reviewer mention not having a place to search for items that could be filtered. Now you have addressed the search box ✅. Please notice your detailed page is currently not loading any data for a specific area. This is among the primary requirements and I strongly suggest you address it. 
> As I see from the error in the console this is caused by using an `HTTP` instead of `HTTPS`, Please consider using `https://api.weatherapi.com/v1/current.json` instead of `http://api.weatherapi.com/v1/current.json`

https://user-images.githubusercontent.com/69077061/221374671-bd3e3017-4dac-4989-a99f-d53289841480.mp4

chiztechnology replied:
hello, dear thanks a lot for your great review and your amazing video.
I really appreciate the fact you took the time to record this video explaining the behavior of the website in the browser.

The app works fine locally, but the hosted version is failing to fetch each weather detail due to some CORS Policy.
kindly have a look at the logs below. 👇 
<img width="1440" alt="Screenshot 2023-02-25 at 22 12 03" src="https://user-images.githubusercontent.com/68948316/221377728-bcebddef-0a4d-445c-8d90-1463e3d7a556.png">

I cleared the codebase as requested by removing all unnecessary commented code.

TadesseDev reviewed (changes requested):
### Hi @chiztechnology 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7](https://user-images.githubusercontent.com/69077061/212385281-4797c58a-d358-423a-aa7c-832c22203865.gif)

## Highlights
 - [ ] Good job overall following the template design ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

chiztechnology committed:
set secure url for api endpoint

chiztechnology committed:
clear code

chiztechnology committed:
Clear code

chiztechnology commented:
@TadesseDev kindly have a look at this previous comment and tell me if there's anything wrong with that.
I think I only have a few remaining reviews to pass this module.

I just wanted to get your idea before requesting for another review

TadesseDev commented:
Sorry I'm not with a computer right now. Make sure to remove all the commented codes and you will be fine. If possible addresse the optional changes too. Good luck 🤞

HAMMAS-SALEEM commented on src/components/Details.js:
```diff
@@ -0,0 +1,104 @@
+        </div>
+
+        <div className="data-grid">
+          {/* Sunrise and Sunset */}
+          <div className="data-grid-item dark">
+
+            <Text>Temperature</Text>
+
+            <Image
+              radius="md"
+              src={obj.current.condition.icon}
+              alt="Random unsplash image"
+              style={{ width: 60, height: 'auto' }}
+            />
+            <div style={{
+              display: 'flex', alignItems: 'center', justifyContent: 'center',
+            }}
+            >
+              <Text style={{ marginBottom: 20, marginTop: 10 }}>
+                It s actually
+                {obj.current.temp_c}
+                °C
+              </Text>
+            </div>
+          </div>
+          {/* Humidity */}
+          <div className="data-grid-item">
+            <Text>Humidity</Text>
+            <Text style={{ fontWeight: 500, fontSize: 42, margin: '30px 0 3px 0' }}>
+              {obj.current.humidity}
+              <span style={{ fontSize: 18 }}>%</span>
+            </Text>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default Details;
```
![image](https://user-images.githubusercontent.com/93039359/221380277-ad149e34-e5ae-4823-817b-397e2725b7f4.png)

**[Optional]**

- [ ] It would be nice if you fix the problem with your details page which was it doesn't show any particular data :fearful: It is highly recommended for you to work on this requirement in order to make your project professional and attractive for the viewers of your repo especially recruiters :+1:

HAMMAS-SALEEM commented on README.md:
```diff
@@ -1 +1,208 @@
+
+
+<!-- SUPPORT -->
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project...
+
```
![image](https://user-images.githubusercontent.com/93039359/221380355-ac621f55-420d-4f69-836c-d05ccf0a93c2.png)

**[Optional]**

- [ ] It would be nice if you add a message here for the viewers of your repo such as **If you like this project then don't forget to give a star :star: to this repo :+1:** to make your readme file descriptive and professional :blush:

HAMMAS-SALEEM reviewed (approved):
# Status: Approved :green_circle: :1st_place_medal: :100:
![success](https://camo.githubusercontent.com/6ada59f6189497c1c2cae6d06f8154b7307a58643f232340e288140ed469eb28/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966)

### Hey @chiztechnology 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Requirements
:heavy_check_mark: Following Gitflow

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
